core/src/main/java/org/axonframework/commandhandling/model/inspection/AnnotatedChildEntity.java

```diff
@@ -71,7 +73,9 @@ public void publish(EventMessage<?> msg, P declaringInstance) {
         if (forwardEvents) {
             Iterable<C> targets = eventTargetResolver.apply(msg, declaringInstance);
             if (targets != null) {
-                targets.forEach(target -> this.entityModel.publish(msg, target));
+                Collection<C> copy = new ArrayList<>();
```

Comment:
I think it would be better to check if `targets` implements `Collection`. In that case, it's faster to create a copy using the copy-constructor on `ArrayList`. That constructor can size the list to the correct size immediately, instead of needing to grow it as the iterator proceeds.
If `target` doesn't implement `Collection`, then there is no alternative than this one...